src/Atmos/Model/AtmosModel.jl

```diff
@@ -275,7 +275,7 @@ function gradvariables!(
     transform.u = ρinv * state.ρu
     transform.h_tot = total_specific_enthalpy(atmos, atmos.moisture, state, aux)
 
-    gradvariables!(atmos.moisture, transform, state, aux, t)
+    gradvariables!(atmos.moisture, atmos, transform, state, aux, t)
```

Comment:
Perhaps set leading argument as `atmos` , followed by subcomponents ?

Reply:
Sure, I'll do that.

Reply:
I think the original reason `atmos` trailed is because `atmos.moisture` is used for dispatch